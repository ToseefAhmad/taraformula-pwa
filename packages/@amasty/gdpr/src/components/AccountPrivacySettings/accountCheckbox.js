import React from 'react';
import { mergeClasses } from '@magento/venia-ui/lib/classify';
import defaultClasses from '../AccountPrivacySettings/accountPrivacySettings.css';
import Checkbox from '@magento/venia-ui/lib/components/Checkbox/checkbox';
import { mustBeChecked } from '@magento/venia-ui/lib/util/formValidators';
import { useIntl } from 'react-intl';
import { func, number, object } from 'prop-types';

const AccountCheckbox = props => {
  const classes = mergeClasses(defaultClasses, props.classes);
  const { formatMessage } = useIntl();
  const { item, index, togglePassword } = props;

  if (!item.hasCheckbox) {
    return null;
  }

  return (
    <div className={classes.checkboxContainer}>
      <Checkbox
        onClick={togglePassword}
        field={`privacy.checkbox${index}`}
        label={formatMessage({
          id: `privacy.checkbox${index}`,
          defaultMessage: item.checkboxText
        })}
        validate={mustBeChecked}
        validateOnBlur
      />
    </div>
  );
};

AccountCheckbox.propTypes = {
  item: object,
  index: number,
  togglePassword: func
};

export default AccountCheckbox;
