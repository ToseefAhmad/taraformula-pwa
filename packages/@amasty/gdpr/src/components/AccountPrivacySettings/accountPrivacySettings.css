.root {
  padding: 0.625rem;
}

.item {
  margin-bottom: 1.25rem;
}

.title {
  border-bottom: 1px solid #000;
  padding-bottom: 0.625rem;
  font-size: 2rem;
  font-weight: 600;
  margin-bottom: 1.5rem;
}

.checkboxContainer,
.password,
.content {
  margin-bottom: 0.625rem;
}
