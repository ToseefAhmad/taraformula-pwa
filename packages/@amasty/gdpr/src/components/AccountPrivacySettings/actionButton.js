import React from 'react';
import { mergeClasses } from '@magento/venia-ui/lib/classify';
import defaultClasses from '../AccountPrivacySettings/accountPrivacySettings.css';
import { FormattedMessage } from 'react-intl';
import Button from '@magento/venia-ui/lib/components/ButtonGroup/button';
import { number, object } from 'prop-types';

const ActionButton = props => {
  const classes = mergeClasses(defaultClasses, props.classes);
  const { item, index } = props;

  if (!item.action) {
    return null;
  }

  return (
    <Button
      classes={{ root_lowPriority: classes.filterButton }}
      disabled={false}
      type="submit"
    >
      <FormattedMessage
        id={`privacy.submitButton${index}`}
        defaultMessage={item.submitText}
      />
    </Button>
  );
};

ActionButton.propTypes = {
  item: object,
  index: number
};

export default ActionButton;
