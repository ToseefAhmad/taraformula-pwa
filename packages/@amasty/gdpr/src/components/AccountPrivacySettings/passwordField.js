import React from 'react';
import { mergeClasses } from '@magento/venia-ui/lib/classify';
import defaultClasses from '../AccountPrivacySettings/accountPrivacySettings.css';
import Password from '@magento/venia-ui/lib/components/Password';
import { isRequired } from '@magento/venia-ui/lib/util/formValidators';
import { useIntl } from 'react-intl';
import { bool, number, object } from 'prop-types';

const PasswordField = props => {
  const classes = mergeClasses(defaultClasses, props.classes);
  const { formatMessage } = useIntl();
  const { item, index, isShowPassword } = props;

  if (!item.needPassword || !item.action || !isShowPassword) {
    return null;
  }

  return (
    <Password
      classes={{ root: classes.password }}
      fieldName="password"
      label={formatMessage({
        id: 'privacy.password' + index,
        defaultMessage: 'Current Password'
      })}
      validate={isRequired}
      isToggleButtonHidden={false}
    />
  );
};

PasswordField.propTypes = {
  item: object,
  index: number,
  isShowPassword: bool
};

export default PasswordField;
