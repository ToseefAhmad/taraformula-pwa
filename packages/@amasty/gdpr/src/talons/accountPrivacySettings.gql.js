import { gql } from '@apollo/client';

export const GET_ACCOUNT_SETTINGS = gql`
  query amGetGdprAccountAction {
    amGetGdprAccountAction {
      title
      content
      hasCheckbox
      checkboxText
      hidePassword
      needPassword
      submitText
      action
      actionCode
    }
  }
`;

export const ACCEPT_GDPR_POLICY = gql`
  mutation acceptAmGdprPolicy($policy_version: String!) {
    acceptAmGdprPolicy(input: { policy_version: $policy_version }) {
      error
      message
    }
  }
`;

export const ANONIMISE_CUSTOMER_DATA = gql`
  mutation anonymiseAmGdprCustomer($password: String!) {
    anonymiseAmGdprCustomer(input: { password: $password }) {
      error
      message
    }
  }
`;

export const DOWNLOAD_CUSTOMER_DATA = gql`
  mutation amGetGdprCustomerPersonalData($password: String!) {
    amGetGdprCustomerPersonalData(input: { password: $password }) {
      label
      value
    }
  }
`;

export const DELETE_CUSTOMER_ACCOUNT = gql`
  mutation requestAmGdprAccountDeletion($password: String!) {
    requestAmGdprAccountDeletion(input: { password: $password }) {
      error
      message
    }
  }
`;

export default {
  getAccountSettings: GET_ACCOUNT_SETTINGS,
  acceptGdprPolicyMutation: ACCEPT_GDPR_POLICY,
  anonimiseCustomerDataMutation: ANONIMISE_CUSTOMER_DATA,
  downloadCustomerDataMutation: DOWNLOAD_CUSTOMER_DATA,
  deleteCustomerAccountMutation: DELETE_CUSTOMER_ACCOUNT
};
