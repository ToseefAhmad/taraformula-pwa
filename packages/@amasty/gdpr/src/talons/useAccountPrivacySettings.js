import { useEffect } from 'react';
import { useQuery } from '@apollo/client';
import { useUserContext } from '@magento/peregrine/lib/context/user';
import { useHistory } from 'react-router-dom';
import mergeOperations from '@magento/peregrine/lib/util/shallowMerge';
import DEFAULT_OPERATIONS from './accountPrivacySettings.gql.js';

export const useAccountPrivacySettings = (props = {}) => {
  const operations = mergeOperations(DEFAULT_OPERATIONS, props.operations);
  const { getAccountSettings } = operations;
  const [{ isSignedIn }] = useUserContext();
  const history = useHistory();

  const { loading, error, data } = useQuery(getAccountSettings, {
    fetchPolicy: 'cache-and-network',
    nextFetchPolicy: 'cache-first',
    skip: !isSignedIn
  });

  // If the user is no longer signed in, redirect to the home page.
  useEffect(() => {
    if (!isSignedIn) {
      history.push('/');
    }
  }, [history, isSignedIn]);

  const { amGetGdprAccountAction: settings } = data || {};

  return {
    settingsLoading: loading,
    error,
    settings
  };
};
