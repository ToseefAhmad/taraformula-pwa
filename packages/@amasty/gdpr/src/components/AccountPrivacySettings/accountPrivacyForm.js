import React from 'react';
import { mergeClasses } from '@magento/venia-ui/lib/classify';
import defaultClasses from '../AccountPrivacySettings/accountPrivacySettings.css';
import { Form } from 'informed';
import AccountCheckbox from './accountCheckbox';
import PasswordField from './passwordField';
import ActionButton from './actionButton';
import AmCheckbox from '../AmCheckbox';
import SuccessMessage from '../SuccessMessage';
import { useAccountPrivacyForm } from '../../talons/useAccountPrivacyForm';
import FormError from '@magento/venia-ui/lib/components/FormError';
import DownloadCsv from '../DownloadCsv/downloadCsv';
import { number, object } from 'prop-types';
import RichContent from '@magento/venia-ui/lib/components/RichContent';

const AccountPrivacyForm = props => {
  const classes = mergeClasses(defaultClasses, props.classes);
  const { item, index } = props;
  const {
    togglePassword,
    isShowPassword,
    handleSubmit,
    errors,
    formSubmitMessage,
    customerData
  } = useAccountPrivacyForm({
    hidePassword: item.hidePassword,
    actionCode: item.actionCode,
    inputName: 'password' + item.actionCode
  });
  const isGivenConsent = item.actionCode === 'given_consents';
  const getConsents = () => {
    return (
      isGivenConsent && (
        <AmCheckbox
          location={'privacy_settings'}
          isAccountSettingsPage={true}
        />
      )
    );
  };

  return (
    <Form key={index} className={classes.item} onSubmit={handleSubmit}>
      <h3 className={classes.title}>{item.title}</h3>
      <RichContent classes={{ root: classes.content }} html={item.content} />
      <AccountCheckbox
        item={item}
        index={index}
        togglePassword={togglePassword}
      />
      <PasswordField
        item={item}
        index={index}
        name={`password${item.actionCode}`}
        isShowPassword={isShowPassword}
      />
      <DownloadCsv data={customerData} />
      {getConsents()}
      <FormError errors={errors} scrollOnError={false} />
      <SuccessMessage message={formSubmitMessage} />
      {!isGivenConsent && !customerData.length && (
        <ActionButton item={item} index={index} />
      )}
    </Form>
  );
};

AccountPrivacyForm.propTypes = {
  item: object,
  index: number
};

export default AccountPrivacyForm;
