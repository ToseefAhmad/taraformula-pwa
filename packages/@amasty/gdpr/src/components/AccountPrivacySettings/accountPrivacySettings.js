import React, { useMemo } from 'react';
import { useAccountPrivacySettings } from '../../talons/useAccountPrivacySettings';
import PolicyButton from '../PolicyButton';
import { fullPageLoadingIndicator } from '@magento/venia-ui/lib/components/LoadingIndicator';
import { mergeClasses } from '@magento/venia-ui/lib/classify';
import defaultClasses from '../AccountPrivacySettings/accountPrivacySettings.css';
import AccountPrivacyForm from './accountPrivacyForm';

const AccountPrivacySettings = props => {
  const classes = mergeClasses(defaultClasses, props.classes);
  const { loading, settings } = useAccountPrivacySettings();

  const settingsBlocks = useMemo(
    () =>
      !Array.isArray(settings) || !settings.length
        ? null
        : settings.map((item, index) => {
            if (!item.action && !item.content) {
              return <PolicyButton key={index} item={item} index={index} />;
            }

            return <AccountPrivacyForm key={index} index={index} item={item} />;
          }),
    [settings]
  );

  if (!settingsBlocks) {
    return null;
  }

  if (loading) {
    return fullPageLoadingIndicator;
  }

  return <div className={classes.root}>{settingsBlocks}</div>;
};

export default AccountPrivacySettings;
