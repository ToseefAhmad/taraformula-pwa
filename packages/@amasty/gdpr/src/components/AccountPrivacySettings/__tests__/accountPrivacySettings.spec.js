import React from 'react';
import { createTestInstance } from '@magento/peregrine';
import AccountPrivacySettings from '../accountPrivacySettings';
import { useAccountPrivacySettings } from '../../../talons/useAccountPrivacySettings';

jest.mock('@magento/venia-ui/lib/components/FormError', () => 'FormError');
jest.mock('@magento/venia-ui/lib/components/RichContent', () => 'RichContent');
jest.mock(
  '@magento/venia-ui/lib/components/ButtonGroup/button',
  () => 'Button'
);
jest.mock('../../PolicyButton', () => 'PolicyButton');
jest.mock('../accountPrivacyForm', () => 'AccountPrivacyForm');
jest.mock('../../../talons/useAccountPrivacySettings', () => {
  const useAccountPrivacySettings = jest.fn(() => {});

  return { useAccountPrivacySettings };
});
jest.mock('@magento/venia-ui/lib/components/LoadingIndicator', () => {
  return {
    fullPageLoadingIndicator: 'LoadingIndicator'
  };
});

test('renders loading', () => {
  useAccountPrivacySettings.mockReturnValue({
    loading: true
  });

  const tree = createTestInstance(<AccountPrivacySettings />);
  expect(tree.toJSON()).toMatchSnapshot();
});

test('renders no settings', () => {
  useAccountPrivacySettings.mockReturnValue({
    loading: false,
    settings: []
  });

  const tree = createTestInstance(<AccountPrivacySettings />);
  expect(tree.toJSON()).toMatchSnapshot();
});

test('renders PolicyButton', () => {
  useAccountPrivacySettings.mockReturnValue({
    loading: false,
    settings: [
      {
        action: false,
        content: false
      }
    ]
  });

  const tree = createTestInstance(<AccountPrivacySettings />);
  expect(tree.toJSON()).toMatchSnapshot();
});

test('renders default', () => {
  useAccountPrivacySettings.mockReturnValue({
    loading: false,
    settings: [
      {
        action: true,
        content: true
      }
    ]
  });

  const tree = createTestInstance(<AccountPrivacySettings />);
  expect(tree.toJSON()).toMatchSnapshot();
});
