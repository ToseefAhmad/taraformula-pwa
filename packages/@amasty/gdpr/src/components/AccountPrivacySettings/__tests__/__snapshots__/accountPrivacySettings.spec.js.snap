// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PolicyButton 1`] = `
<div>
  <PolicyButton
    index={0}
    item={
      Object {
        "action": false,
        "content": false,
      }
    }
  />
</div>
`;

exports[`renders default 1`] = `
<div>
  <AccountPrivacyForm
    index={0}
    item={
      Object {
        "action": true,
        "content": true,
      }
    }
  />
</div>
`;

exports[`renders loading 1`] = `null`;

exports[`renders no settings 1`] = `null`;
